### PR TITLE
[rllib] Refactor to support passing custom env_creator function

### PR DESCRIPTION
**Pull request by ericl**
cc @richardliaw @pcmoritz

### CODE REVIEW

File: python/ray/rllib/ppo/ppo.py [hunk 1]
```diff
@@ -60,7 +60,7 @@
     # If >1, adds frameskip
     "extra_frameskip": 1,
     # Number of timesteps collected in each outer loop
-    "timesteps_per_batch": 40000,
```
Comment (pcmoritz) [inline python/ray/rllib/ppo/ppo.py]: needs to be changed back

**Review by pcmoritz — COMMENTED** [review]
Also, there is some linting that needs to be fixed.

File: python/ray/rllib/ppo/ppo.py (hunk 1 above)
Comment (ericl) [reply python/ray/rllib/ppo/ppo.py]: IMO we should keep it at 4k, 40k is very likely to crash any laptop user using this script, as it does each time I run PPO by accident :)

For tuned params, they can refer to the examples.

**Comment by ericl** [comment]
Fixed lint

File: python/ray/rllib/common.py [hunk 2]
```diff
@@ -118,28 +122,35 @@ class Agent(object):
...
         """
+        self._experiment_id = uuid.uuid4().hex
         upload_dir = "file:///tmp/ray" if upload_dir is None else upload_dir
-        self.experiment_id = uuid.uuid4().hex
-        self.env_name = env_name
+        if type(env_creator) is str:
+            env_name = env_creator
+            self.env_creator = lambda: gym.make(env_name)
+        else:
+            env_name = "custom"
```
Comment (richardliaw) [inline python/ray/rllib/common.py]: is this desirable?

**Comment by ericl** [comment]
So there are a couple alternatives: the user could provide a name, or we
could construct the env and pull it's spec.id.

**Comment by pcmoritz** [comment]
Ok, let's merge this for now, we can think about names for custom MDPs as a followup.

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**